src/components/SingleTask.tsx

```diff
@@ -1,29 +1,101 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Draggable } from 'react-beautiful-dnd';
 import { AiFillDelete, AiFillEdit } from 'react-icons/ai';
-import { MdDone } from 'react-icons/md';
+import { MdDone, MdClear } from 'react-icons/md';
 
 interface Props {
+  index: number;
   task: string;
   isCompleted: boolean;
+  todos: string[];
+  setTodos: React.Dispatch<React.SetStateAction<string[]>>;
+  completedTodos: string[];
+  setCompletedTodos: React.Dispatch<React.SetStateAction<string[]>>;
 }
 
-const SingleTask: React.FC<Props> = ({ task, isCompleted }) => {
+const SingleTask: React.FC<Props> = ({ index, task, isCompleted, todos, setTodos, completedTodos, setCompletedTodos }) => {
```

Comment:
```suggestion
const SingleTask: React.FC<Props> = ({ index, task, isCompleted, todos, setTodos, completedTodos, setCompletedTodos }) => {
```
Can we instead only passing either `todos and setTodos` for active tasks or `completed and setCompletedTodos` for completed tasks?